walk-in-interview/src/main/java/com/google/job/data/JobPayment.java

```diff
@@ -5,20 +5,27 @@
     private final int min;
     private final int max;
     private final PaymentFrequency paymentFrequency;
+    private final long annualMax;
+
+    private static final int HOURS_PER_YEAR = 8760;
+    /* Note that is an approximate value */
+    private static final int WEEKS_PER_YEAR = 52;
+    private static final int MONTHS_PER_YEAR = 12;
 
     private volatile int hashCode;
 
     // For serialization
     public JobPayment() {
-        this(/* min= */0, /* max= */0, PaymentFrequency.HOURLY);
+        this(/* min= */0, /* max= */0, PaymentFrequency.HOURLY, /* annualMax= */ 0);
     }
 
-    public JobPayment(int min, int max, PaymentFrequency paymentFrequency) {
-        validateParameter(min, max);
+    public JobPayment(int min, int max, PaymentFrequency paymentFrequency, long annualMax) {
```

Comment:
FYI, after this param you should not add any more params to this class. if you need to add another one, you should change to a Builder pattern JobPayment.Builder

Reply:
noted, thanks